boskos/metrics/metrics.go

```diff
@@ -39,33 +40,39 @@ var (
 	promMetrics = prometheusMetrics{
 		BoskosState: map[string]map[string]prometheus.Gauge{},
 	}
-	resources common.ResTypes
+	resources, states common.ResTypes
+	defaultStates     = []string{
+		common.Busy,
+		common.Cleaning,
+		common.Dirty,
+		common.Free,
+		common.Leased,
+	}
 )
 
 func init() {
 	flag.Var(&resources, "resource-type", "comma-separated list of resources need to have metrics collected")
+	flag.Var(&states, "resource-state", "comma-separated list of states need to have metrics collected")
+	flag.Parse()
```

Comment:
eh, leave flag.Parse() in main, and append to states afterwards?

Reply:
why ? from doc you should call flag.Parse once you have defined all your flags.

Reply:
what if later on there are multiple init calls in the same package?